src/ios/CDVNotification.m

```diff
@@ -228,9 +228,19 @@ - (void)beep:(CDVInvokedUrlCommand*)command
 
 -(UIViewController *)getTopPresentedViewController {
     UIViewController *presentingViewController = self.viewController;
-    while(presentingViewController.presentedViewController != nil && ![presentingViewController.presentedViewController isBeingDismissed])
-    {
-        presentingViewController = presentingViewController.presentedViewController;
+    if (presentingViewController.view.window != [UIApplication sharedApplication].keyWindow){
+        //for wkwebview, the privacy screen plugin is presented from a different window object
+        presentingViewController = [UIApplication sharedApplication].keyWindow.rootViewController;
+        while (presentingViewController.presentedViewController != nil && ![presentingViewController.presentedViewController isBeingDismissed]){
+            presentingViewController = presentingViewController.presentedViewController;
+        }
+    }
+    else {
+        //for uiwebview viewcontroller, if inappbrowser or privacyscreen viewcontroller is presented,
+        //then they should be used to repsent the new view control.
+        while (presentingViewController.presentedViewController != nil && ![presentingViewController.presentedViewController isBeingDismissed]){
```

Comment:
This while is the same as in the if, so you can remove the else and put the while outside the if.
Also, the comments don't make much sense, can you delete them?

Reply:
The while clause will be moved out the if-else block; and the whole else block will be removed. 

Should I keep the below comment in the if block? 
 "//for wkwebview, the privacy screen plugin is presented from a different window object"

Jonathan

Reply:
I don't really know what do you mean by "privacy screen plugin", that's why I said that it didn't make much sense.
The problem is about InAppBrowser plugin, right? In InAppBrowser plugin, it's presented in a different window for both UIWebView and WKWebView.
If you want to put a comment, you can do it before the if and say, something like "Some plugins might be presented from a different window than the app window" or something like that, but I think the code is clear enough.

Reply:
Sure, I will remove the comment in a new commit.

Jonathan